### PR TITLE
Updates to complier doc

### PR DESCRIPTION
**Pull request by Ehud-Sharlin**

### CODE REVIEW

File: docs/reference/rocmcc/rocmcc.md [hunk 1]
```diff
@@ -3,21 +3,892 @@
 ROCmCC is a Clang/LLVM-based compiler. It is optimized for high-performance computing on AMD GPUs and CPUs and supports various heterogenous programming models such as HIP, OpenMP, and OpenCL.
```
Comment (saadrahim) [inline docs/reference/rocmcc/rocmcc.md:3]: The title should be Compiler Reference Guide.
A section with ```##``` should start with Introduction

File: docs/reference/rocmcc/rocmcc.md [hunk 2]
```diff
@@ -3,21 +3,892 @@
...
 
 ROCmCC is made available via two packages: rocm-llvm and rocm-llvm-alt. The differences are shown in this table:
-
+||
+|:--:|
 | <b>Table 1. rocm-llvm vs. rocm-llvm-alt</b>|
```
Comment (saadrahim) [inline docs/reference/rocmcc/rocmcc.md:8]: Please do not use embedded HTML. Please see https://www.markdownguide.org/basic-syntax#bold

File: docs/reference/rocmcc/rocmcc.md [hunk 3]
```diff
@@ -3,21 +3,892 @@
...
+| Device function inlining | Inlines all GPU device functions, which provide greater performance and compatibility for codes that contain file scoped or device function scoped \_\_shared\_\_ variables. However, it may increase compile time. | Relies on inlining heuristics to control inlining. Users experiencing performance or compilation issues with code using file scoped or device function scoped \_\_shared\_\_ variables could try -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false to work around the issue. There are plans to address these issues with future compiler improvements. |
+| Source code location | Developed at [https://github.com/ROCm-Developer-Tools/HIPCC](https://github.com/ROCm-Developer-Tools/HIPCC) | Developed at [https://github.com/RadeonOpenCompute/llvm-project](https://github.com/RadeonOpenCompute/llvm-project) |
+
+
+
+# Compiler Options and Features
+
```
Comment (saadrahim) [inline docs/reference/rocmcc/rocmcc.md:54]: Please do not use Level 1 headings except for the page title

**Review by saadrahim — CHANGES_REQUESTED** [review]
Please see comments for minor changes.

File: docs/reference/rocmcc/rocmcc.md (hunk 1 above)
Comment (Ehud-Sharlin) [reply docs/reference/rocmcc/rocmcc.md:5]: Fixed, moved all titles one step down the hierarchy

File: docs/reference/rocmcc/rocmcc.md (hunk 2 above)
Comment (Ehud-Sharlin) [reply docs/reference/rocmcc/rocmcc.md]: Fixed, will need to remove similar HTML leftovers from magma_install.md

File: docs/reference/rocmcc/rocmcc.md (hunk 3 above)
Comment (Ehud-Sharlin) [reply docs/reference/rocmcc/rocmcc.md:56]: Fixed, moved all titles one step down the hierarchy

Comment (Ehud-Sharlin) [reply docs/reference/rocmcc/rocmcc.md:56]: Please note that rocmcc is still a work-in-progress draft, still working on a few missing parts and cleaning table 9. Thanks

**Review by saadrahim — APPROVED** [review]

**Pull request merged**